crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -569,14 +573,33 @@ pub fn find_dot_flox(
         debug!(".flox found: path={}", tentative_dot_flox.display());
         return Ok(Some(pointer));
     }
-    // Search upwards for a .flox
-    while let Some(grandparent) = tentative_dot_flox.parent().and_then(|p| p.parent()) {
-        let grandparent_clone = grandparent.to_path_buf();
-        tentative_dot_flox = grandparent.join(DOT_FLOX);
+
+    // Check if we're in a git repo.
+    let toplevel =
+        match GitCommandProvider::toplevel(&path).map_err(EnvironmentError2::DetectGitDirectory)? {
+            None => return Ok(None),
+            Some(toplevel) => toplevel,
+        };
```

Comment:
Rather than having a somewhat separate Git interface I'd rather use the Repo based one we already have:

```
let toplevel = match Git::discover(".") {
	Some(repo) if repo.workdir().is_some() => repo.workdir.unwrap()
	Some(_) | None| Err(GitError::NotFound) => return Ok(None)
	Err(e) => return Err(EnvironmentError2::DetectGitDirectory)
}
```

Reply:
Are you using `GitError::NotFound` as a placeholder for `Err(GitCommandError::BadExit(128, _, _))`? Are you suggesting we change `discover`, or actually use as is? I don't think the caller should have to match on something like `Err(GitCommandError::BadExit(128, _, _))`

Reply:
I mean 

```
Err(e) if e.not_found() => return Ok(None)
Err(e) => EnvironmentError2::DetectGitDirectory 
```


generally our error/return types for some git methods need some work but funnily enough discover is one of the methods with a specialized type

Reply:
Okay I dropped toplevel and matched how you suggest. I'm not sure if `not_found` is swallowing other errors, but I guess we can address that if it comes up 🤷